app/order-by.pipe.ts

```diff
@@ -0,0 +1,18 @@
+import {Pipe, PipeTransform} from '@angular/core';
+import {Course} from './course'
+
+@Pipe({
+    name: 'orderByDate'
```

Comment:
The requirement was to create universal `orderBy` pipe that would accept the parameter by which to order. You have coupled your pipe with your domain logic. This is not good for app architecture.